frontend/src/lib/components/PasswordStrength.tsx

```diff
@@ -1,26 +1,58 @@
+import { LemonDivider } from '@posthog/lemon-ui'
 import { LemonProgress } from 'lib/lemon-ui/LemonProgress'
 import { Tooltip } from 'lib/lemon-ui/Tooltip'
 import zxcvbn from 'zxcvbn'
 
+export type ValidatedPasswordResult = {
+    score: number // 0 is no passsword - otherwise 1-5,
+    feedback?: string
+}
+
+export function validatePassword(password: string = ''): ValidatedPasswordResult {
+    // Checks the validation against the zxcvbn library
+    // and any other custom validation we have
+
+    const result = zxcvbn(password)
+
+    if (result.score > 3 && password.length < 8) {
+        return {
+            score: 3,
+            feedback: 'Must be at least 8 characters long',
+        }
+    }
+
+    return {
+        score: password ? result.score + 1 : 0,
+        feedback: result.feedback.suggestions.join(' '),
+    }
+}
+
 export default function PasswordStrength({
-    password = '',
-    className,
+    validatedPassword,
 }: {
-    password?: string
-    className?: string
+    validatedPassword: ValidatedPasswordResult
 }): JSX.Element {
-    // passwordScore is 0 if no password input
-    // passwordScore is 20, 40, 60, 80, or 100 if password input, based on zxcvbn score (which is 0, 1, 2, 3, or 4)
-    const passwordScore: number = password.length && zxcvbn(password).score * 20 + 20
+    const { score, feedback } = validatedPassword
 
     return (
-        <Tooltip title="Password strength">
+        <Tooltip
```

Comment:
Nit: the tooltip hover area is very small and I'm struggling to get it to show up even knowing it's there. Not a blocker but we may want to try and make the hover surface area larger.